code/clue.js

```diff
@@ -1,59 +1,374 @@
+  article: 'a',
+  id: 'bat',
+  foundIn: 'billiardRoom'
+}
+
+const trophy = {
+  name: 'trophy',
+  weight: 29,
+  article: 'a',
+  id: 'trophy',
+  foundIn: 'livingRoom'
+}
+
+const pistol = {
+  name: 'pistol',
+  weight: 11,
+  article: 'a',
+  id: 'pistol',
+  foundIn: 'lounge'
+}
+
+//objects for rooms
+const diningRoom = {
+  name: 'dining Room',
+  id: 'dining room'
+}
+
+const conservatory = {
+  name: 'conservatory',
+  id: 'conservatory'
+}
+
+const kitchen = {
+  name: 'kitchen',
+  id: 'kitchen'
+}
+
+const study = {
+  name: 'study',
+  id: 'study'
+}
+
+const library = {
+  name: 'library',
+  id: 'library'
+}
+
+const billiardRoom = {
+  name: 'billiard Room',
+  id: 'billiard Room'
+}
+
+const lounge = {
+  name: 'lounge',
+  id: 'lounge'
+}
+
+const ballroom = {
+  name: 'ballroom',
+  id: 'ballroom'
+}
+
+const hall = {
+  name: 'hall',
+  id: 'hall'
+}
+
+const spa = {
+  name: 'spa',
+  id: 'spa'
 }
 
-// THE ROOMS ONLY HAS A NAME SO NO NEED FOR OBJECTS THERE.
+const livingRoom = {
+  name: 'living Room',
+  id: 'living Room'
+}
+
+const observatory = {
+  name: 'observatory',
+  id: 'observatory'
+}
+
+const theater = {
+  name: 'theater',
+  id: 'theater'
+}
 
-// NOW GROUP ALL SUSPECTS, WEAPONS AND ROOMS IN ARRAYS LIKE THIS:
+const guestHouse = {
+  name: 'guest House',
+  id: 'guest House'
+}
 
+const patio = {
+  name: 'patio',
+  id: 'patio'
+}
+ 
 const suspects = [
   mrGreen,
+  professorPlum,
+  missScarlet,
+  mrsPeacock,
+  colonelMustard,
   mrsWhite
-  // ...  and the rest
 ]
 
-const weapons = []
+const weapons = [
+  rope,
+  knife,
+  candelstick,
+  dumbbell,
+  poison,
+  axe,
+  bat,
+  trophy,
+  pistol
+]
 
-const rooms = []
+const rooms = [
+  diningRoom, 
+  conservatory, 
+  kitchen, 
+  study, 
+  library, 
+  billiardRoom, 
+  lounge, 
+  ballroom, 
+  hall, 
+  spa, 
+  livingRoom, 
+  observatory, 
+  theater, 
+  guestHouse, 
+  patio
+]
 
 // THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
-// YOU DON'T NEED TO CHANGE THIS, JUST TRY TO UNDERSTAND IT. AND HOW TO USE IT.
 const randomSelector = array => {
   return array[Math.floor(Math.random() * array.length)]
 }
 
-// CREATE AN OBJECT THAT KEEPS THE MYSTERY.
-// With a killer, a weapon and a room.
-// The values will be set later.
+//object that keeps mystery 
+const mystery = {
+  killer: null,
+  weapon: null,
+  room: null
+};
 
-// This function will be invoked when you click on the killer card.
+// This function will be invoked when you click the killer card.
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  if (!killerIsPicked) {
+    document.getElementById('loaderKiller').style.opacity = 1
+    setTimeout(function () {
+      document.getElementById('loaderKiller').style.opacity = 0
+      shuffleFavoriteWeapon();
+      mystery.killer = randomSelector(suspects)
```

Comment:
you are forgeting about semicolons from time to time

Reply:
Thanks for the input, I have updated semicolons.